src/f32-vbinary/vopc-rvv.c.in

```diff
@@ -0,0 +1,69 @@
+// Copyright 2023 SiFive, Inc.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+$assert LMUL >= 1
+$assert OP in ["ADD", "DIV", "RDIV", "MAX", "MIN", "MUL", "SUB", "RSUB", "SQRDIFF"]
+$assert ACTIVATION in ["LINEAR", "MINMAX", "RELU"]
+#include <assert.h>
+
+#include <riscv_vector.h>
+
+#include <xnnpack/common.h>
+#include <xnnpack/vbinary.h>
+
+
+$OP_FUNC = {
+$  "ADD": "__riscv_vfadd_vf_f32",
+$  "DIV": "__riscv_vfdiv_vf_f32",
+$  "RDIV": "__riscv_vfrdiv_vf_f32",
+$  "MAX": "__riscv_vfmax_vf_f32",
+$  "MIN": "__riscv_vfmin_vf_f32",
+$  "MUL": "__riscv_vfmul_vf_f32",
+$  "SUB": "__riscv_vfsub_vf_f32",
+$  "RSUB": "__riscv_vfrsub_vf_f32",
+$  "SQRDIFF": "__riscv_vfsub_vf_f32",
+$}[OP]
+$SUFFIX = {"LINEAR": "", "RELU": "_relu", "MINMAX": "_minmax"}[ACTIVATION]
+$PARAMS = {"LINEAR": "xnn_f32_default_params", "RELU": "xnn_f32_relu_params", "MINMAX": "xnn_f32_minmax_params"}[ACTIVATION]
+void xnn_f32_v${OP.lower()}c${SUFFIX}_ukernel__rvv_u${LMUL}v(
+    size_t batch,
+    const float* input_a,
+    const float* input_b,
+    float* output,
+    const union ${PARAMS} params[restrict XNN_MIN_ELEMENTS(1)])
+{
+  assert(batch != 0);
+  assert(batch % sizeof(float) == 0);
+  assert(input_a != NULL);
+  assert(input_b != NULL);
+  assert(output != NULL);
+
+  size_t n = batch >> 2;
+  float b = *input_b;
+  $if ACTIVATION == "MINMAX":
+    size_t vl = __riscv_vsetvl_e32m${LMUL}(n);
+    vfloat32m${LMUL}_t voutput_min = __riscv_vfmv_v_f_f32m${LMUL}(params->scalar.min, vl);
+    vfloat32m${LMUL}_t voutput_max = __riscv_vfmv_v_f_f32m${LMUL}(params->scalar.max, vl);
+  $elif ACTIVATION == "RELU":
+    size_t vl = __riscv_vsetvl_e32m${LMUL}(n);
+    vfloat32m${LMUL}_t voutput_min = __riscv_vfmv_v_f_f32m${LMUL}(0.0f, vl);
+
+  do {
+    size_t vl = __riscv_vsetvl_e32m${LMUL}(n);
```

Comment:
consider moving this to bottom of loop and ensure the first one is done outside of loop for MINMAX/RELU and else (LINEAR)